andela/ah-backend-thanos
# #161348797 Users should be able to create articles

#### What does this PR do?
Create an Article and Read all Articles

#### Description of Task to be completed?
- Creates a new article with required fields (title, description, body, image_url, and tag_list)
- Automatically generates a slug based on the title
- Creates a unique slug if a similar one already exists by adding a timestamp
- Retrieves all articles and show user details for each article
- Tests for getting all articles and creating an article
- Adds validation for article fields

#### How should this be manually tested?
- Run the server: `python manage.py runserver`
- In postman, use the endpoints: 
GET `api/users/login` to login and get the token
GET `api/articles` to get all articles
POST `api/articles` to post an article with the following fields:

HEADERS: 
Key: `Authorization`  Value: `Token <token value>`

Payload:
```
{
    "title": "How to train your dragon",
    "description": "Ever wonder how?",
    "body": "It takes a Jacobian",
    "image_url": "http://iviidev.info/downloads/image.jpg",
    "tag_list": ["science", "technology"]
}
```

Expected Result:
```
{
    "article": {
        "id": 1,
        "slug": "how-to-train-your-dragon",
        "title": "How to train your dragon",
        "description": "Ever wonder how?",
        "body": "It takes a Jacobian",
        "tag_list": [
            "science",
            "technology"
        ],
        "created_at": "2018-11-01T03:14:21.547439Z",
        "updated_at": "2018-11-01T03:14:21.547460Z",
        "favorited": false,
        "favorites_count": 0,
        "image_url": "http://iviidev.info/downloads/image.jpg",
        "audio_url": null,
        "author": {
            "id": 1,
            "username": "brucemakallan",
            "email": "brucemakallan@gmail.com"
        }
    }
}
```

#### Any background context you want to provide?
You require a login token to Post an Article. To get that token, you need to sign up AND verify your email address

#### What are the relevant pivotal tracker stories?
#161348797 Users should be able to create articles

https://www.pivotaltracker.com/story/show/161348797
